src/operator/src/statement/copy_database.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use common_datasource::file_format::Format;
+use common_datasource::lister::{Lister, Source};
+use common_datasource::object_store::build_backend;
+use common_query::Output;
+use common_telemetry::{debug, error, info, tracing};
+use object_store::Entry;
+use regex::Regex;
+use session::context::{QueryContextBuilder, QueryContextRef};
+use snafu::{ensure, ResultExt};
+use table::requests::{CopyDatabaseRequest, CopyDirection, CopyTableRequest};
+
+use crate::error;
+use crate::error::{CatalogSnafu, InvalidCopyParameterSnafu};
+use crate::statement::StatementExecutor;
+
+pub(crate) const COPY_DATABASE_TIME_START_KEY: &str = "start_time";
+pub(crate) const COPY_DATABASE_TIME_END_KEY: &str = "end_time";
+pub(crate) const CONTINUE_ON_ERROR_KEY: &str = "continue_on_error";
+
+impl StatementExecutor {
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn copy_database_to(&self, req: CopyDatabaseRequest) -> error::Result<Output> {
+        // location must end with / so that every table is exported to a file.
+        ensure!(
+            req.location.ends_with('/'),
+            InvalidCopyParameterSnafu {
+                key: "location",
+                value: req.location,
+            }
+        );
+
+        info!(
+            "Copy database {}.{} to dir: {}, time: {:?}",
+            req.catalog_name, req.schema_name, req.location, req.time_range
+        );
+        let table_names = self
+            .catalog_manager
+            .table_names(&req.catalog_name, &req.schema_name)
+            .await
+            .context(CatalogSnafu)?;
+
+        let suffix = Format::try_from(&req.with)
+            .context(error::ParseFileFormatSnafu)?
+            .suffix();
+
+        let mut exported_rows = 0;
+        for table_name in table_names {
```

Comment:
It cannot handle the tables of Metric engine yet, we need to add a TODO and support it in the future.

Reply:
If we are to copy metric engine tables, we must find a way to save and restore the table metadata, too. For now, we need to filter out the metric engine tables.

Reply:
It's better to create an issue cc @waynexia

Reply:
Added a todo here https://github.com/GreptimeTeam/greptimedb/commit/1da7d2e6b7e02f972b92d52e23d821af060a1361#diff-b3f149877e58dcda21272ac161833e8417f156d325d0d74465f7fc5f0f70a8a5R68